corehq/apps/app_manager/views/apps.py

```diff
@@ -947,7 +947,7 @@ def require_remote_app():
     if should_edit("custom_suite"):
         app.set_custom_suite(hq_settings['custom_suite'])
 
-    return HttpResponse(json.dumps(resp))
+    return HttpResponse(json.dumps(resp), content_type='application/json')
```

Comment:
Are there advantages/disadvantages to switching this and similar responses to a `JsonResponse`?

Reply:
`JsonResponse` sets the content-type automatically and also does the serialization for us, which is great. On the other hand, for much of the serialization we do, we'd have to set a `safe=False` flag on it, which to me felt less explicit than setting the `content_type` on an `HttpResponse`. 

I see both used throughout HQ, though there does seem to be a preference for `JsonResponse()` over `HttpResponse(json.dumps())`, so maybe that's the route I should go.

Reply:
Interesting, I didn't know the bit about `safe`. Defer to you on the best approach here.

Reply:
I think you're right that `JsonResponse` is a little bit nicer for most cases. Made that change (although the linter isn't happy with something now. https://github.com/dimagi/commcare-hq/pull/35110/commits/5e88e043618386dc9688739d7b45aa15d9a79fa0

Reply:
Sweet, thank you. And now I have a bunch of questions about when to use `safe` 🙃 My assumption is that `safe` should be false whenever the response contains user-generated data.

Reply:
It looks like prior to ES5 (so ~2013), there was a potential vulnerability with passing arrays as JSON. `safe` here seems to refer to that vulnerability specifically, so in practice `safe=False` should always be fine. https://stackoverflow.com/a/70204451

Reply:
excellent